File: frontend/src/services/stats.ts
import api from './api';
import { DashboardStats } from '../types';

export const getDashboardStats = async (): Promise<DashboardStats> => {
  const response = await api.get<DashboardStats>('/stats');
  return response.data;
};

export const getRSVPBreakdown = async (): Promise<Record<string, number>> => {
  const response = await api.get('/stats/rsvp-breakdown');
  return response.data;
};

export const getSideBreakdown = async (): Promise<Record<string, number>> => {
  const response = await api.get('/stats/side-breakdown');
  return response.data;
};

export const getRelationBreakdown = async (): Promise<Record<string, number>> => {
  const response = await api.get('/stats/relation-breakdown');
  return response.data;
};

File: backend/app/routers/stats.py
from fastapi import APIRouter, Depends
from sqlalchemy.ext.asyncio import AsyncSession

from app.database import get_db
from app.services.stats_service import StatsService
from app.schemas.common import StatsResponse

router = APIRouter(prefix="/stats", tags=["Statistics"])


@router.get("", response_model=StatsResponse)
async def get_dashboard_stats(db: AsyncSession = Depends(get_db)):
    """Get dashboard statistics."""
    service = StatsService(db)
    stats = await service.get_dashboard_stats()
    return stats


@router.get("/rsvp-breakdown")
async def get_rsvp_breakdown(db: AsyncSession = Depends(get_db)):
    """Get RSVP status breakdown."""
    service = StatsService(db)
    breakdown = await service.get_rsvp_breakdown()
    return breakdown


@router.get("/side-breakdown")
async def get_side_breakdown(db: AsyncSession = Depends(get_db)):
    """Get guest side breakdown."""
    service = StatsService(db)
    breakdown = await service.get_side_breakdown()
    return breakdown


@router.get("/relation-breakdown")
async def get_relation_breakdown(db: AsyncSession = Depends(get_db)):
    """Get guest relation breakdown."""
    service = StatsService(db)
    breakdown = await service.get_relation_breakdown()
    return breakdown
